src/Uno.UI/UI/Xaml/Controls/Control/Control.cs

```diff
@@ -166,9 +166,29 @@ private protected void SuppressIsEnabled(bool suppress)
 			}
 		}
 
-		private object CoerceIsEnabled(object baseValue)
+		private protected virtual object CoerceIsEnabled(object baseValue, DependencyPropertyValuePrecedences precedence)
 		{
-			return _suppressIsEnabled ? false : baseValue;
+			if (_suppressIsEnabled)
+			{
+				return false;
+			}
+
+			// The baseValue hasn't been set yet, so we need to make sure we're not
+			// reading soon-to-be-outdated values
+			var localValue = precedence < DependencyPropertyValuePrecedences.Inheritance ? // < actually means higher precedence
+				baseValue :
+				((IDependencyObjectStoreProvider)this).Store.GetPropertyDetails(IsEnabledProperty).GetValueUnderPrecedence(DependencyPropertyValuePrecedences.Coercion).value;
```

Comment:
`localValue` is evaluated too early. It could get evaluated without actually being used. We might also want to avoid calling `GetPropertyDetails` twice (and also avoid calling it unnecessarily)